tests/cli/dbt_project.yml

```diff
@@ -0,0 +1,5 @@
+name: jaffle_shop
+profile: jaffle_shop
+version: '0.1.0'
+config-version: 2
+clean-targets: [target, dbt_packages, logs]
```

Comment:
What is the motivation for defining the project files in this manner as opposed to what's in place in terms of fixtures for the existing functional tests? 

For example, this file could be overwritten in an individual test method or class via the [dbt_project_yaml fixture](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/tests/fixtures/project.py#L180). The existing fixtures have some nice test isolation + cleanup functionality that should get rid of the need to [.gitignore `dbt_packages` in this PR](https://github.com/dbt-labs/dbt-core/pull/6723/files#diff-25a58f2510df1a432759522298c0ec0fc19b09b9e432947fc4369a7e823f87aeR1) and make it more straightforward to write additional cli tests that have their own isolated project environment.